distributedsocialnetwork/post/views.py

```diff
@@ -33,7 +34,20 @@ def create_post(request):
 
 def view_post(request, pk):
     context = {}
+    if request.POST:
+        form = CommentCreationForm(request.POST)
+        if form.is_valid():
+            new_comment = form.save(commit=False)
+            new_comment.author_id = user.id
```

Comment:
Since this is in a different view, user is undefined. We just have to set request.user=user first.

Reply:
This is corrected as well now, thanks.